pytorch_lightning/trainer/lr_finder.py

```diff
@@ -0,0 +1,288 @@
+        model.configure_optimizers = lr_finder._get_new_optimizer(optimizer[0])
+
+        # Fit, lr & loss logged in callback
+        self.fit(model)
+
+        # Promt if we stopped early
+        if self.global_step != num_iters:
+            print('LR finder stopped early due to diverging loss.')
+
+        # Transfer results from callback to lr finder object
+        lr_finder.results.update({'lr': self.callbacks[0].lrs,
+                                  'loss': self.callbacks[0].losses})
+
+        # Finish by resetting variables so trainer is ready to fit model
+        self.logger = logger
+        self.callbacks = callbacks
+        self.max_steps = max_steps
+        self.show_progress_bar = show_progress_bar
+        self.accumulate_grad_batches = accumulate_grad_batches
+        model.configure_optimizers = configure_optimizers
+
+        return lr_finder
+
+
+class _LRFinder(object):
+    def __init__(self, mode, lr_min, lr_max, num_iters):
+        assert mode in ('linear', 'exponential'), \
+            'mode should be either `linear` or `exponential`'
+
+        self.mode = mode
+        self.lr_min = lr_min
+        self.lr_max = lr_max
+        self.num_iters = num_iters
+
+        self.results = {}
+
+    def _get_new_optimizer(self, optimizer):
+        new_lrs = [self.lr_min] * len(optimizer.param_groups)
+        for param_group, new_lr in zip(optimizer.param_groups, new_lrs):
+            param_group["lr"] = new_lr
+            param_group["initial_lr"] = new_lr
+
+        args = (optimizer, self.lr_max, self.num_iters)
+        scheduler = _LinearLR(*args) if self.mode == 'linear' else _ExponentialLR(*args)
+
+        def configure_optimizers():
+            return [optimizer], [{'scheduler': scheduler,
+                                  'interval': 'step'}]
+
+        return configure_optimizers
+
+    def plot(self, suggest=False, ax=None):
+        import matplotlib.pyplot as plt
+
+        lrs = self.results["lr"]
+        losses = self.results["loss"]
+
+        fig = None
+        if ax is None:
+            fig, ax = plt.subplots()
+
+        # Plot loss as a function of the learning rate
+        ax.plot(lrs, losses)
+        if self.mode == 'exponential':
+            ax.set_xscale("log")
+        ax.set_xlabel("Learning rate")
+        ax.set_ylabel("Loss")
+
+        if suggest:
+            _ = self.suggestion()
+            if self._optimal_idx:
+                ax.plot(lrs[self._optimal_idx], losses[self._optimal_idx],
+                        markersize=10, marker='o', color='red')
+
+        if fig is not None:
+            plt.show()
+
+        return ax
+
+    def suggestion(self):
+        try:
+            min_grad = (np.gradient(np.array(self.results["loss"]))).argmin()
+            self._optimal_idx = min_grad
+            return self.results["lr"][min_grad]
+        except Exception:
+            print('Failed to compute suggesting for lr.'
+                  'There might not be enough points.')
```

Comment:
```suggestion
            log.warning('Failed to compute suggesting for `lr`. There might not be enough points.')
```
and import `from pytorch_lightning import _logger as log`